sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -116,17 +121,25 @@ public boolean requiresDeduping() {
      * <ul>
      * <li>This finalize method may be called from any thread, concurrently with calls to
      * the {@link UnboundedReader} it was created from.
+     * <li>This method may be called on a {@link CheckpointMark} that has been decoded by the
```

Comment:
Throughout, there's a tonal ambiguity between "may be" as in "is permitted to be" and "maybe" as in "you don't know".